rules/manager_test.go

```diff
@@ -192,8 +192,8 @@ func TestAlertingRule(t *testing.T) {
 }
 
 func TestForStateAddSamples(t *testing.T) {
-	for _, evalDelay := range []time.Duration{0, time.Minute} {
-		t.Run(fmt.Sprintf("evalDelay %s", evalDelay.String()), func(t *testing.T) {
+	for _, queryOffset := range []time.Duration{0, time.Minute} {
+		t.Run("queryOffset "+queryOffset.String(), func(t *testing.T) {
```

Comment:
Am I missing something or this is different than upstream? Checking prometheus `main` I can see:

```suggestion
		t.Run(fmt.Sprintf("queryOffset %s", queryOffset.String()), func(t *testing.T) {
```

Reply:
You're right - I fixed all the linter failures blindly and these were one of them (it basically complains of using `fmt.Sprintf` instead of string concatenation when possible).

I took a brief look and I can see that our setup in CI is slightly different from the one in Prometheus. `golangcilint` configuration files are definitely the same but the way they are used is different - I suspect the problem with the CI being skipped/not-check lies in upstream Prometheus and not here. 

EDIT: I figured it out, the upgrade of golangcilint broke this linter: https://github.com/prometheus/prometheus/pull/14215

I can do either of two things:

1. I can fix it in Prometheus and then come back.
2. I can upgrade the version of `golangcilint` here so that it matches prometheus and then I can safely remove the changes.

Reply:
>I can upgrade the version of golangcilint here so that it matches prometheus and then I can safely remove the changes.

I would do this. We aim to be on par with prometheus as much as possible.

Reply:
I have:

1. Reverted the linter fixes with https://github.com/grafana/mimir-prometheus/pull/639/commits/3836920aa9d20dfd14cef050346ea23d855c5d3d and https://github.com/grafana/mimir-prometheus/pull/639/commits/2e6cc9b06eeac9822c7f83f01c12240b8474bab1
2. Upgraded `golangcilint` to match Prometheus with https://github.com/grafana/mimir-prometheus/pull/639/commits/9cce125267604599fa927479e23235c471d25241
3. and finally, fixed 4 lints from changes that are not in Prometheus https://github.com/grafana/mimir-prometheus/pull/639/commits/199a070f2bfe78c131d45274706c18a2f152e1b0